Library/Homebrew/cmake/trap_fetchcontent_provider.cmake

```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.24) # Dependency providers introduced in CMake 3.24
+
+option(HOMEBREW_ALLOW_FETCHCONTENT "Allow FetchContent to be used in Homebrew builds" OFF)
+
+macro(trap_fetchcontent_provider method depName)
+    if (NOT HOMEBREW_ALLOW_FETCHCONTENT)
```

Comment:
I wonder if we can/should just return early if this is set.

Reply:
I'm open to this and it is possible, but I don't think it's as idiomatic in CMake (nor does it offer as much of a readability benefit as it does in Ruby):

```cmake
if (HOMEBREW_ALLOW_FETCHCONTENT)
    return()
endif()

macro(trap_fetchcontent_provider method depName)
    [...]
endif()
```